sanskrit_parser/parser/sandhi.py

```diff
@@ -422,7 +424,7 @@ def main():
             args.join = True
         with outputctx(args.strict_io):
             if args.split:
-                word_in = SanskritObject(args.word, encoding=ie, strict_io=args.strict_io)
+                word_in = SanskritNormalizedString(args.word, encoding=ie, strict_io=args.strict_io)
```

Comment:
This is different from the documentation of the functions, which specify that the input params should be `SanskritImmutableString`s. `SanskritNormalizedString` is not derived from `SanskritImmutableString`. I know that Python uses duck typing, so this won't cause any errors, but I think this makes it confusing. Should we rethink the class hierarchy?

Reply:
We should indeed. I started off on trying to get a more flexible hierarchy going, but didn't end up with a coherent one. I've also started thinking of a sanskrit_parser.generator, and wanted the string hierarchy to support that as well (that's one reason I thought of mutable vs immutable). However, I don't think my result is cogent. This is definitely a candidate for (yet another) revamp.